samples/bluetooth/broadcast_audio_assistant/src/main.c

```diff
@@ -0,0 +1,520 @@
+						 sr_info.broadcast_name)) {
+					printk("Match found for '%s'\n", CONFIG_SELECT_SOURCE_NAME);
+				} else {
+					printk("'%s' not found in names\n\n",
+					       CONFIG_SELECT_SOURCE_NAME);
+					return;
+				}
+			}
+
+			err = bt_le_scan_stop();
+			if (err != 0) {
+				printk("bt_le_scan_stop failed with %d\n", err);
+			}
+
+			/* TODO: Add support for syncing to the PA and parsing the BASE
+			 * in order to obtain the right subgroup information to send to
+			 * the sink when adding a broadcast source (see in main function below).
+			 */
+
+			printk("Selecting Broadcast ID: 0x%06x\n", sr_info.broadcast_id);
+
+			selected_broadcast_id = sr_info.broadcast_id;
+			selected_sid = info->sid;
+			selected_pa_interval = info->interval;
+			bt_addr_le_copy(&selected_addr, info->addr);
+
+			k_sem_give(&sem_source_discovered);
+		}
+	} else {
+		/* Scan for and connect to Broadcast Sink */
+
+		/* We are only interested in connectable advertisers */
+		if ((info->adv_props & BT_GAP_ADV_PROP_CONNECTABLE) == 0) {
+			return;
+		}
+
+		bt_data_parse(ad, device_found, (void *)&sr_info);
+
+		if (sr_info.has_bass && sr_info.has_pacs) {
+			printk("Broadcast Sink Found:\n");
+			printk("  BT Name:        %s\n", sr_info.bt_name);
+
+			if (strlen(CONFIG_SELECT_SINK_NAME) > 0U) {
+				/* Compare names with CONFIG_SELECT_SINK_NAME */
+				if (is_substring(CONFIG_SELECT_SINK_NAME, sr_info.bt_name)) {
+					printk("Match found for '%s'\n", CONFIG_SELECT_SINK_NAME);
+				} else {
+					printk("'%s' not found in names\n\n",
+					       CONFIG_SELECT_SINK_NAME);
+					return;
+				}
+			}
+
+			err = bt_le_scan_stop();
+			if (err != 0) {
+				printk("bt_le_scan_stop failed with %d\n", err);
+			}
+
+			printk("Connecting to Broadcast Sink: %s\n", sr_info.bt_name);
+
+			err = bt_conn_le_create(info->addr, BT_CONN_LE_CREATE_CONN,
+						BT_LE_CONN_PARAM_DEFAULT,
+						&broadcast_sink_conn);
+			if (err != 0) {
+				printk("Failed creating connection (err=%u)\n", err);
+				scan_for_broadcast_sink();
+			}
+
+			k_sem_give(&sem_sink_discovered);
+		}
+	}
+}
+
+static void scan_timeout_cb(void)
+{
+	printk("Scan timeout\n");
+}
+
+static struct bt_le_scan_cb scan_callbacks = {
+	.recv = scan_recv_cb,
+	.timeout = scan_timeout_cb,
+};
+
+static void scan_for_broadcast_source(void)
+{
+	int err;
+
+	scanning_for_broadcast_source = true;
+
+	err = bt_le_scan_start(BT_LE_SCAN_PASSIVE, NULL);
+	if (err) {
+		printk("Scanning failed to start (err %d)\n", err);
+		return;
+	}
+
+	printk("Scanning for Broadcast Source successfully started\n");
+
+	err = k_sem_take(&sem_source_discovered, K_FOREVER);
+	if (err != 0) {
+		printk("Failed to take sem_source_discovered (err %d)\n", err);
+	}
+}
+
+static void scan_for_broadcast_sink(void)
+{
+	int err;
+
+	scanning_for_broadcast_source = false;
+
+	err = bt_le_scan_start(BT_LE_SCAN_PASSIVE, NULL);
+	if (err) {
+		printk("Scanning failed to start (err %d)\n", err);
+		return;
```

Comment:
Does it make sense that we change this to have a return value, and abort the main function if it fails? Or continue the main loop in that case

Reply:
if it fails scanning, there is something broken in the main logic - I've never seen it happen.  Right now it will just print to inform the user.